### PR TITLE
131 add adaptive layout

### PR DESCRIPTION
**Pull request by poludnev**

### CODE REVIEW

File: src/components/TweetResults/components/NotarizeButton/NotarizeButton.tsx [hunk 1]
```diff
@@ -27,59 +27,71 @@ export const NotarizeButton: React.FC<INotarizeButtonProps> = () => {
...
-        navigate(`/results`);
         dispatch(hideModal);
-        setTransactionId(result.transactionHash!);
-        setTransactionStatus('success');
-      }, 1500);
+      }, 3000);
+      return;
+    }
+
+    updateStateOnTransaction('Waiting for transaction...');
```
Comment (utgarda) [inline src/components/TweetResults/components/NotarizeButton/NotarizeButton.tsx:72]: https://www.figma.com/file/E9Fe71isrR7uFiFdt3Yw7b/Finity-Design-System-%F0%9F%92%9C-(Community)?type=design&node-id=1537-43950&mode=design&t=qttRyNX7qMCeD4DW-0


image

Pending transaction

File: src/components/TweetResults/components/NotarizeButton/NotarizeButton.tsx [hunk 2]
```diff
@@ -27,59 +27,71 @@ export const NotarizeButton: React.FC<INotarizeButtonProps> = () => {
...
+    console.log(`Notatize button on click, tweetId: ${tweetId}, nftMetadataCid: ${contentId?.nftMetadataCid}`);
+    const result = await submitNotarization(tweetId, contentId?.nftMetadataCid, updateStateOnTransaction);
+    console.log(result);
+    if (result.status === 'failed') {
+      updateStateOnTransaction(result.error ? result.error : 'Transaction declined');
+
+      setTimeout(() => dispatch(hideModal), 3000);
```
Comment (utgarda) [inline src/components/TweetResults/components/NotarizeButton/NotarizeButton.tsx:79]: A constant or an .env entry will be handy for timeouts

File: src/components/TweetResults/components/TweetStats/TweetStats.tsx [hunk 3]
```diff
@@ -7,37 +7,38 @@ export const TweetStats: React.FC<ITweetStatsProps> = ({
   views_count,
   bookmark_count,
 }) => {
+  console.log(favorite_count, quote_count, retweet_count, views_count, bookmark_count);
```
Comment (utgarda) [inline src/components/TweetResults/components/TweetStats/TweetStats.tsx:10]: .

File: src/contexts/ContractContext.tsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  useContractEvent({
+    address: notaryShotContract.address as `0x${string}`,
+    abi: notaryShotContract.abi,
+    eventName: 'Transfer',
+    listener([log]) {
+      console.log('Contract on Transfer Event, log:', log);
+
+      const { args } = log as typeof log & { args: { tokenId: bigint; to: `0x${string}` } };
+
+      args.tokenId.toString();
```
Comment (utgarda) [inline src/contexts/ContractContext.tsx:35]: .

File: src/contexts/ContractContext.tsx [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+  nftId: null,
+};
+
+export const ContractContext = createContext<IContractContext>(contractContextInitValue);
+
+export const useContractContext = () => useContext(ContractContext);
+
+export const ContractContextProvider = ({ children }: { children: React.ReactNode }): JSX.Element => {
+  const [nftId, setNftId] = useState<string | null>(null);
+  const { address } = useAccount();
```
Comment (utgarda) [inline src/contexts/ContractContext.tsx:24]: Can we rename this to `userAddress` here? So many `address` words all around

File: src/index.tsx [hunk 6]
```diff
@@ -50,25 +41,41 @@ const router = createBrowserRouter([
   { path: '*', element: <Page404 /> },
 ]);
 
+const { chains, publicClient } = configureChains([polygon], [publicProvider()]);
+
+const { connectors } = getDefaultWallets({
+  appName: 'Quantum Oracle App',
+  projectId: '6c1f221e5a67ca1b6d9ecc2ec3303b18',
```
Comment (utgarda) [inline src/index.tsx:48]: Move projectId to build variables

File: src/lib/apiClient.ts [hunk 7]
```diff
@@ -65,28 +64,30 @@ export const submitNotarization = async (
   tweetId: string,
   cid: string,
   cb?: (data: string) => void,
-): Promise<
-  // TODO: https://github.com/orgs/NotarizedScreenshot/projects/1/views/1?filterQuery=typ&pane=issue&itemId=28111890
-  ContractReceipt | { status: 'failed' | 'success'; transactionHash?: string | null; error?: string | null }
-> => {
+): Promise<{ status: 'failed' | 'success'; transactionHash?: string | null; error?: string | null }> => {
```
Comment (utgarda) [inline src/lib/apiClient.ts:67]: `transactionHash?: string | null;` 
` error?: string | null`
seems a bit messy

**Review by utgarda — APPROVED** [review]
Pls take care of the nits,
invite another pair of eyes to see all the scss,
 and merge!

**Review by habdevs — CHANGES_REQUESTED** [review]
I was mostly guided by rem recommendations based on past experience, so maybe my rem was exposed erroneously as not everywhere there is a need for it.

File: src/components/Benefits/Benefits.module.scss [hunk 8]
```diff
@@ -3,24 +3,78 @@
   .h2 {
     max-width: 92rem;
     margin-bottom: 2.8rem;
+    @media screen and (max-width: 1023px) {
```
[the remark this answers is not in the document]
Comment (poludnev) [reply src/components/Benefits/Benefits.module.scss:6]: Why should we? What advantages does it give to us? 

I'm not sure about the behavior of that query on a "print" type device so use it implicitly for the "screen".

File: src/components/Header/Header.module.scss [hunk 9]
```diff
@@ -6,8 +6,111 @@
...
+        height: 2px;
+        border-radius: 100px;
+        &::before,
+        &::after {
+          position: absolute;
+          content: '';
+          width: 3rem;
+          height: 2px;
+          background-color: inherit;
+          border-radius: 100px;
```
[the remark this answers is not in the document]
Comment (poludnev) [reply src/components/Header/Header.module.scss:105]: rem or em are useless for such kind of styles as radius, when you want border be rounded.

**Pull request merged**